flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java

```diff
@@ -154,4 +149,149 @@ public void testFromGenericOptionsParser() throws IOException {
+
+		// test parameter access
+		Assert.assertEquals(4.0, parameter.getFloat("float"), 0.00001);
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+
+		// test repeated access
+		Assert.assertEquals(4.0, parameter.getFloat("float"), 0.00001);
+	}
+
+	@Test
+	public void testUnrequestedDouble() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"-double", "8"});
+		Assert.assertEquals(Sets.newHashSet("double"), parameter.getUnrequestedParameters());
+
+		// test parameter access
+		Assert.assertEquals(8.0, parameter.getDouble("double"), 0.00001);
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+
+		// test repeated access
+		Assert.assertEquals(8.0, parameter.getDouble("double"), 0.00001);
+	}
+
+	@Test
+	public void testUnrequestedString() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"-string", "∞"});
+		Assert.assertEquals(Sets.newHashSet("string"), parameter.getUnrequestedParameters());
+
+		// test parameter access
+		Assert.assertEquals("∞", parameter.get("string"));
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+
+		// test repeated access
+		Assert.assertEquals("∞", parameter.get("string"));
+	}
+
+	@Test
+	public void testUnrequestedHas() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"-boolean"});
+		Assert.assertEquals(Sets.newHashSet("boolean"), parameter.getUnrequestedParameters());
+
+		// test parameter access
+		Assert.assertTrue(parameter.has("boolean"));
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+
+		// test repeated access
+		Assert.assertTrue(parameter.has("boolean"));
+	}
+
+	@Test
+	public void testUnrequestedRequired() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"-required", "∞"});
+		Assert.assertEquals(Sets.newHashSet("required"), parameter.getUnrequestedParameters());
+
+		// test parameter access
+		Assert.assertEquals("∞", parameter.getRequired("required"));
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+
+		// test repeated access
+		Assert.assertEquals("∞", parameter.getRequired("required"));
+	}
+
+	@Test
+	public void testUnrequestedUnknown() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{});
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+
+		Assert.assertTrue(parameter.getBoolean("boolean", true));
+		Assert.assertEquals(0, parameter.getByte("byte", (byte) 0));
+		Assert.assertEquals(0, parameter.getShort("short", (short) 0));
+		Assert.assertEquals(0, parameter.getInt("int", 0));
+		Assert.assertEquals(0, parameter.getLong("long", 0));
+		Assert.assertEquals(0, parameter.getFloat("float", 0), 0.00001);
+		Assert.assertEquals(0, parameter.getDouble("double", 0), 0.00001);
+		Assert.assertEquals("0", parameter.get("string", "0"));
+	}
```

Comment:
* how about also testing more complex scenarios with multiple (existing) parameters?
* or adding tests that verify the non-default, i.e. `getRequired()`, versions of the accessors - independent of the `RuntimeException` being thrown, the `unrequestedParamaters` field should correctly reflect the requests
* what happens for
```
ParameterTool parameter = ParameterTool.fromArgs(new String[]{"int"});
try {
  parameter.getInt("int");
} catch (RuntimeException e) {
  Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
}
```